astropy/io/misc/yaml.py

```diff
@@ -377,5 +377,5 @@ def dump(data, stream=None, **kwargs):
 
     """
     kwargs["Dumper"] = AstropyDumper
-    kwargs.setdefault("default_flow_style", None)
+    kwargs.setdefault("sort_keys", False)
```

Comment:
Why did you remove the `default_flow_style` default? That's an unrelated setting, although it's unclear to me if this has now made the tests still fail...

Reply:
Sorry, that wasn't supposed to happen. Thanks for noticing, I'll make the correct changes now as you suggested.